### DISCUSSION TITLE
How to do fit and test at the same time with Lightning CLI ?

### DISCUSSION THREAD

**Discussion by Hannibal046** [Lightning-AI/pytorch-lightning#17300] [Lightning Trainer API: Trainer, LightningModule, LightningDataModule] [answered]
Hi, I am wondering how to `fit` and `test` my model with a single CLI?
Currently, i am just using:
```
python main.py fit ...
python main.py test ckpt_path
```
This is not very convenient for me since and the log dir is unknown before `fit`(v_num),so i cannot wrap the two commands above into a script. And i have to find the best ckpt_path myself. Is there any better workaround here ?

**Comment by mauvilsa** [comment, marked as answer]
Instead of having a CLI with subcommands, you can use the [instantiation only mode](https://lightning.ai/docs/pytorch/stable/cli/lightning_cli_advanced_3.html#instantiation-only-mode) and call test right after fit.

However, a fair warning. The test set should be used as few times as possible. Measuring performance on the test set too often is a bad practice because you end up optimizing on the test. So, technically it is better to use the test subcommand giving explicitly a checkpoint (only one among many you may have) and not plan to run the test for every fit you do.
